CRM/Utils/Mail/EmailProcessor.php

```diff
@@ -241,16 +251,20 @@ public static function _process($civiMail, $dao, $is_create_activities) {
           if (!empty($dao->activity_status)) {
             $params['status_id'] = $dao->activity_status;
           }
-          $result = civicrm_api('activity', 'create', $params);
 
-          if ($result['is_error']) {
-            $matches = FALSE;
-            echo "Failed Processing: {$mail->subject}. Reason: {$result['error_message']}\n";
-          }
-          else {
-            $matches = TRUE;
-            CRM_Utils_Hook::emailProcessor('activity', $params, $mail, $result);
-            echo "Processed as Activity: {$mail->subject}\n";
+          // Create activity if its not empty.
+          $matches = FALSE;
+          if (!empty($params['subject']) || !empty($params['target_contact_id'])) {
```

Comment:
I think I'm still not clear on this part. Suppose you turn on the setting to skip creating contacts. So $params['target_contact_id'] will be empty. Suppose also that the `from` address is already a contact in civi. And suppose also that it's an email with no subject but it has a body. It should file that email, right?

Reply:
Okay will revert this for now.

Reply:
Thanks